terraform/cloudwatch.tf.json.template.py

```diff
@@ -7,28 +7,48 @@
 
 emit_tf(None if config.disable_monitoring else {
     "resource": [
-        {
-            "aws_cloudwatch_metric_alarm": {
-                f"{lambda_}_5xx": {
-                    "alarm_name": config.qualified_resource_name(lambda_ + '_5xx'),
-                    "comparison_operator": "GreaterThanThreshold",
-                    # This alarm will catch persistent 5XX errors occurring over
-                    # one hour, specifically when more than one occurrence is
-                    # sampled in a ten-minute period for six consecutive periods.
-                    "evaluation_periods": 6,
-                    "period": 60 * 10,
-                    "metric_name": "5XXError",
-                    "namespace": "AWS/ApiGateway",
-                    "statistic": "Sum",
-                    "threshold": 1,
-                    "treat_missing_data": "notBreaching",
-                    "dimensions": {
-                        "ApiName": config.qualified_resource_name(lambda_),
-                        "Stage": config.deployment_stage,
+        *(
+            {
+                "aws_cloudwatch_metric_alarm": {
+                    f"{lambda_}_5xx": {
+                        "alarm_name": config.qualified_resource_name(lambda_ + '_5xx'),
+                        "comparison_operator": "GreaterThanThreshold",
+                        # This alarm will catch persistent 5XX errors occurring over
+                        # one hour, specifically when more than one occurrence is
+                        # sampled in a ten-minute period for six consecutive periods.
+                        "evaluation_periods": 6,
+                        "period": 60 * 10,
+                        "metric_name": "5XXError",
+                        "namespace": "AWS/ApiGateway",
+                        "statistic": "Sum",
+                        "threshold": 1,
+                        "treat_missing_data": "notBreaching",
+                        "dimensions": {
+                            "ApiName": config.qualified_resource_name(lambda_),
+                            "Stage": config.deployment_stage,
+                        },
+                        "alarm_actions": ["${aws_sns_topic.azul_monitoring.arn}"]
                     }
                 }
             }
+            for lambda_ in config.lambda_names()
+        ),
+        {
+            'aws_sns_topic': {
+                'azul_monitoring': {
+                    'name': 'azul-monitoring'
+                }
+            },
+            'aws_sns_topic_subscription': {
+                'azul_monitoring_azul_group': {
```

Comment:
As a rule of thumb, don't encode the value of an artifact in the name of said artifact. This is similar to a previous PR where you encoded the value of a variable in its name. The point of an abstraction, like a variable or a method or a TF resource is that you can change the implementation (or the value) without affecting the parts of the system referencing that abstraction. If you encode the value of a variable in its name, you have to change the name of the variable if you change the value. If you change the name, you have to change all places referencing that variable and you've gained nothing over not having a variable in the first place.